### PR TITLE
Fork tests for NFTDescriptor to check compatibility with v2.0 and v2.1

### PR DESCRIPTION
**Pull request by smol-ninja**
Closes https://github.com/sablier-labs/v2-core/issues/881 and https://github.com/sablier-labs/v2-core/issues/890

### CODE REVIEW

File: src/SablierV2NFTDescriptor.sol [hunk 1]
```diff
@@ -78,6 +78,13 @@ contract SablierV2NFTDescriptor is ISablierV2NFTDescriptor {
             })
         );
 
+        // Performs a low-level call to handle v2.0 contracts in which `isTransferable` is not implemented.
+        (bool success, bytes memory returnData) =
+            address(vars.sablier).staticcall(abi.encodeWithSignature("isTransferable(uint256)", streamId));
```
Comment (PaulRBerg) [inline src/SablierV2NFTDescriptor.sol:83]: There's a nicer way to do this with [`abi.encodeCall`](https://twitter.com/PaulRBerg/status/1494307965729480704).

**Review by PaulRBerg — APPROVED** [review]
lgtm, but please review my changes @smol-ninja before merging

File: test/fork/NFTDescriptor.t.sol [hunk 2]
```diff
@@ -48,11 +48,11 @@ abstract contract NFTDescriptor_Fork_Test is Fork_Test {
     /// @dev Given enough fuzz runs, all the following scenarios will be fuzzed:
     /// - Multiple values of streamId.
     function testForkFuzz_TokenURI_LockupDynamic_V2_0(uint256 streamId) external loadDeployments_V2_0 {
-        streamId = bound(streamId, 1, lockupDynamic.nextStreamId() - 1);
+        streamId = _bound(streamId, 1, lockupDynamic.nextStreamId() - 1);
```
Comment (smol-ninja) [inline test/fork/NFTDescriptor.t.sol:51]: Why `_bound` and not `bound`?

**Review by smol-ninja — COMMENTED** [review]
Thanks for the review @PaulRBerg and improving the comments. They look better now.

File: test/fork/NFTDescriptor.t.sol (hunk 2 above)
Comment (PaulRBerg) [reply test/fork/NFTDescriptor.t.sol:51]: Doesn't emit console logs.

https://github.com/foundry-rs/forge-std/blob/e4aef94c1768803a16fe19f7ce8b65defd027cfd/src/StdUtils.sol#L60-L63

File: test/fork/NFTDescriptor.t.sol [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+
+        // Set the new NFT descriptor for the previous version of Lockup Linear.
+        resetPrank({ msgSender: lockupLinear.admin() });
+        lockupLinear.setNFTDescriptor(nftDescriptor);
+
+        string memory tokenURIAfter = lockupLinear.tokenURI(streamId);
+
+        // Assert that the token URI has not changed.
+        assertEq(tokenURIBefore, tokenURIAfter, "incompatible token URI");
+    }
```
[the remark this answers is not in the document]
Comment (smol-ninja) [reply test/fork/NFTDescriptor.t.sol:97]: Oh shoot! You're right.

[the remark this answers is not in the document]
Comment (PaulRBerg) [reply test/fork/NFTDescriptor.t.sol:97]: Opened an issue for tracking: https://github.com/sablier-labs/v2-core/issues/893

**Pull request merged**